dissect/target/plugins/os/windows/tasks.py

```diff
@@ -58,452 +61,132 @@
+        "restart_on_failure_interval",
+        "restart_on_failure_count",
+        "mutiple_instances_policy",
+        "dissalow_start_on_batteries",
+        "stop_going_on_batteries",
+        "start_when_available",
+        "network_profile_name",
+        "run_only_network_available",
+        "wake_to_run",
+        "enabled",
+        "hidden",
+        "delete_expired_task_after",
+        "idle_duration",
+        "idle_wait_timeout",
+        "idle_stop_on_idle_end",
+        "idle_restart_on_idle",
+        "network_settings_name",
+        "network_settings_id",
+        "execution_time_limit",
+        "priority",
+        "run_only_idle",
+        "unified_scheduling_engine",
+        "disallow_start_on_remote_app_session",
+        "data",
+        "_target",
+    ]
 
     PATHS = [
         "sysvol/windows/system32/tasks",
         "sysvol/windows/system32/tasks_migrated",
         "sysvol/windows/syswow64/tasks",
+        "sysvol/windows/tasks",  # at.exe job file location
     ]
 
-    def __init__(self, target):
+    def __init__(self, target: Target):
         super().__init__(target)
-        self.files = []
+        self.task_objects = []
 
-        for path in self.PATHS:
-            fpath = self.target.fs.path(path)
+        for file_path in self.PATHS:
+            fpath = self.target.fs.path(file_path)
             if not fpath.exists():
                 continue
 
             for entry in fpath.rglob("*"):
-                if entry.is_file():
-                    self.files.append(entry)
+                if entry.is_file() and entry.suffix.lower() == ".job":
+                    self.task_objects.append(AtTask(entry, target))
+                elif entry.is_file() and not entry.suffix:
+                    self.task_objects.append(XmlTask(entry, target))
 
     def check_compatible(self):
-        if len(self.files) == 0:
+        if len(self.task_objects) == 0:
             raise UnsupportedPluginError("No task files")
 
-    @export(record=DynamicDescriptor(["uri", "datetime"]))
-    def tasks(self):
+    @export(record=DynamicDescriptor(["path", "datetime"]))
+    def tasks(self) -> Iterator:
         """Return all scheduled tasks on a Windows system.
 
         On a Windows system, a scheduled task is a program or script that is executed on a specific time or at specific
         intervals. An adversary may leverage such scheduled tasks to gain persistence on a system.
 
         References:
-            - https://en.wikipedia.org/wiki/Windows_Task_Scheduler
+            https://en.wikipedia.org/wiki/Windows_Task_Scheduler
+
+        Yields:
+            The scheduled tasks found on the target.
         """
-        for f in self.files:
+        for task_object in self.task_objects:
             try:
-                for entry in self.parse_task(f):
+                for entry in self.get_task_fields(task_object):
                     yield entry
             except Exception as e:
```

Comment:
Isn't this Exception a bit too wide? What kind of exceptions do you expect. If nothing, you can remove the exception handling completely because then the test has to fail so we can detect the anomaly. Of course there can be other reasons for this handler that I cannot see, in that case, please add a comment to explain.